### ISSUE TITLE
NPE when reading file

### ISSUE THREAD

**Issue by stevespringett** [spdx/tools#88]
I'm attempting to use the spdxviewer and the verify tool for the first time. Both result in a null pointer exception being thrown.

```bash
$ java -jar spdx-tools-2.1.4-jar-with-dependencies.jar SPDXViewer myfile.spdx 
log4j:WARN No appenders could be found for logger (Jena).
log4j:WARN Please initialize the log4j system properly.
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.spdx.rdfparser.SPDXDocumentFactory.createSpdxDocument(SPDXDocumentFactory.java:104)
	at org.spdx.tools.SpdxViewer.main(SpdxViewer.java:61)
	at org.spdx.tools.Main.main(Main.java:27)
Caused by: java.lang.NullPointerException
	at org.apache.jena.tdb.sys.EnvTDB.processGlobalSystemProperties(EnvTDB.java:33)
	at org.apache.jena.tdb.TDB.init(TDB.java:248)
	at org.apache.jena.tdb.sys.InitTDB.start(InitTDB.java:29)
	at org.apache.jena.system.JenaSystem.lambda$init$2(JenaSystem.java:119)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.apache.jena.system.JenaSystem.forEach(JenaSystem.java:194)
	at org.apache.jena.system.JenaSystem.forEach(JenaSystem.java:171)
	at org.apache.jena.system.JenaSystem.init(JenaSystem.java:117)
	at org.apache.jena.util.FileManager.<clinit>(FileManager.java:86)
	... 3 more
```
```bash
$ java -jar spdx-tools-2.1.4-jar-with-dependencies.jar Verify myfile.spdx 
log4j:WARN No appenders could be found for logger (Jena).
log4j:WARN Please initialize the log4j system properly.
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.spdx.rdfparser.SPDXDocumentFactory.createSpdxDocument(SPDXDocumentFactory.java:104)
	at org.spdx.tools.CompareSpdxDocs.openRdfOrTagDoc(CompareSpdxDocs.java:1152)
	at org.spdx.tools.Verify.main(Verify.java:51)
	at org.spdx.tools.Main.main(Main.java:49)
Caused by: java.lang.NullPointerException
	at org.apache.jena.tdb.sys.EnvTDB.processGlobalSystemProperties(EnvTDB.java:33)
	at org.apache.jena.tdb.TDB.init(TDB.java:248)
	at org.apache.jena.tdb.sys.InitTDB.start(InitTDB.java:29)
	at org.apache.jena.system.JenaSystem.lambda$init$2(JenaSystem.java:119)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.apache.jena.system.JenaSystem.forEach(JenaSystem.java:194)
	at org.apache.jena.system.JenaSystem.forEach(JenaSystem.java:171)
	at org.apache.jena.system.JenaSystem.init(JenaSystem.java:117)
	at org.apache.jena.util.FileManager.<clinit>(FileManager.java:86)
	... 4 more
```

The file I'm using for input was generated by spdx-maven-plugin v0.2.5 which is the latest release in Maven Central.

**Comment by goneall** [comment]
@stevespringett you might try version 2.1.2 or earlier.  We recently updated the version of Jena and it looks like the NPE is originating in that recently updated code.  Also check that your are running Java 8 which is a new dependency introduced with the update Jena.

If the SPDX file is not confidential and not too large, could you upload it to this issue and I'll see if I can reproduce?

@yevster Let me know if you have any thoughts on the bug

I did find a couple stack overflow articles on the NPE (e.g. http://stackoverflow.com/questions/36755846/jena-tdb-java-lang-exceptionininitializererror)

**Comment by stevespringett** [comment]
Thanks for the response. I tried 2.1.0 and it works as expected. The SPDXViewer works, and Verify states what was generated from the maven plugin is valid.

FYI, the project I was using as a test is: https://github.com/stevespringett/dependency-track/tree/3.0-dev . I'm looking to add SPDX import/export functionality to that project.

Attached is the output of running 0.2.5 maven plugin

[Dependency-Track-3.0.0-SNAPSHOT.spdx.zip](https://github.com/spdx/tools/files/856122/Dependency-Track-3.0.0-SNAPSHOT.spdx.zip)

**Comment by goneall** [comment]
I found a reference to what I believe this issue is: https://mail-archives.apache.org/mod_mbox/jena-users/201602.mbox/%3C56B47B8A.2040407@apache.org%3E

I was able to duplicate the problem by building the jar files with Maven then executing the JAR files outside the Eclipse IDE (e.g. java -jar ...)

Fixed by using the Maven shade plugin to control the building of the dependencies.  Unfortunately, this surfaced some potential class name overlaps - not sure if they will cause any issues, but I'll file a separate issue to track.
